packages/realm/src/app-services/App.ts

```diff
@@ -278,25 +279,31 @@ export class App<
       assert.function(fetch, "fetch");
     }
 
+    const syncClientConfig: binding.SyncClientConfig_Relaxed = {
+      multiplexSessions,
+      userAgentBindingInfo: App.userAgent,
+    };
+
     fs.ensureDirectoryForFile(fs.joinPaths(baseFilePath || fs.getDefaultDirectoryPath(), "mongodb-realm"));
     // TODO: This used getSharedApp in the legacy SDK, but it's failing AppTests
-    this.internal = binding.App.getApp(
-      binding.AppCacheMode.Disabled,
-      {
-        appId: id,
-        deviceInfo: App.deviceInfo,
-        transport: createNetworkTransport(fetch),
-        baseUrl,
-        defaultRequestTimeoutMs: timeout ? binding.Int64.numToInt(timeout) : undefined,
-      },
-      {
-        baseFilePath: baseFilePath ? baseFilePath : fs.getDefaultDirectoryPath(),
-        metadataMode: metadata ? toBindingMetadataMode(metadata.mode) : binding.MetadataMode.NoEncryption,
-        customEncryptionKey: metadata?.encryptionKey,
-        userAgentBindingInfo: App.userAgent,
-        multiplexSessions,
-      },
-    );
+    this.internal = binding.App.getApp(binding.AppCacheMode.Disabled, {
+      appId: id,
+      deviceInfo: App.deviceInfo,
+      transport: createNetworkTransport(fetch),
+      baseUrl,
+      defaultRequestTimeoutMs: timeout ? binding.Int64.numToInt(timeout) : undefined,
+      baseFilePath: baseFilePath ? baseFilePath : fs.getDefaultDirectoryPath(),
+      metadataMode: metadata ? toBindingMetadataMode(metadata.mode) : binding.MetadataMode.NoEncryption,
+      customEncryptionKey: metadata?.encryptionKey,
+      syncClientConfig,
```

Comment:
I'd probably inline this, if there's no need for the extra local const (which also skips the need to declare the type).

```suggestion
      syncClientConfig: {
        multiplexSessions,
        userAgentBindingInfo: App.userAgent,
      }
```